src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -14,13 +14,11 @@ namespace Microsoft.DocAsCode.Build.TableOfContents
 
     internal sealed class TocResolver
     {
-        private readonly IHostService _host;
         private readonly Dictionary<string, TocItemInfo> _collection;
         private readonly Dictionary<FileAndType, TocItemInfo> _notInProjectTocCache = new Dictionary<FileAndType, TocItemInfo>();
 
-        public TocResolver(IHostService host, Dictionary<string, TocItemInfo> collection)
+        public TocResolver(Dictionary<string, TocItemInfo> collection)
```

Comment:
DocFx tries to provide backward compatibility for public APIs. For this change, we can mark `public TocResolver(IHostService host, Dictionary<string, TocItemInfo> collection) : this(collection) { }` as `[Obsolete]`, and add another constructor removing the unneeded parameter.

Same for other public API signature changes.

Reply:
but that class is internal

Reply:
what are the other public APIs i changed?